### PR TITLE
Developer Documentation removed from Documentation Page, Page Reordered

### PR DESCRIPTION
**Pull request by JohnMarksRH**
Changes proposed in this pull request:

- Developer Documentation Deleted from main Documentation Page

- Other sections on page reordered to accommodate the change

- All links have been tested

- Middleman screenshot attached 
(https://cloud.githubusercontent.com/assets/25613123/25813250/805fd9c6-3421-11e7-9ab6-c687759e3294.png)

I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md) @jmarks

This pull request needs review by: (please @duck-rh @doron-fediuck @dankenigsberg @mykaul

### CODE REVIEW

File: source/documentation/index.html.haml [hunk 1]
```diff
@@ -58,51 +58,22 @@ wiki_last_updated: 2014-04-26
...
+      - [Introduction](./introduction)
+      - [Administration](./admin-guide)
+      - [Architecture](./architecture)
+      - [How To](./how-to)
+      - [Internal](./internal)
+      - [Security](./security)
+      - [SLA](./sla)
+      - [Storage](./storage)
+      - [Drafts](./draft)
+      - [ּresources](./resources)
```
Comment (duck-rh) [inline source/documentation/index.html.haml]: Capital R has transformed into a middle dot and small r

File: source/documentation/index.html.haml [hunk 2]
```diff
@@ -119,19 +90,31 @@ wiki_last_updated: 2014-04-26
...
-      -   [Storage](./storage)
-      -   [Drafts](./draft)
-      -   [Resources](./resources)
+      ### Getting Involved
+
+      Do you want to help us improve the documentation for oVirt?
+
+      - Sign up for a [GitHub](https://github.com) account, and log in
+      - Help fix some of the [issues](http://github.com/ovirt/ovirt-site/issues) we've identified
+      - Help out with some [website gradening](https://github.com/oVirt/ovirt-site)
```
Comment (duck-rh) [inline source/documentation/index.html.haml]: typo on "gardening", which was not in original version (copy-paste problem?)

File: source/documentation/index.html.haml (hunk 1 above)
Comment (JohnMarksRH) [inline source/documentation/index.html.haml]: @duck-rh Thanks, what do you mean by 'middle dot'?

File: source/documentation/index.html.haml (hunk 2 above)
Comment (JohnMarksRH) [inline source/documentation/index.html.haml]: @duck-rh good catch thanks

File: source/documentation/index.html.haml (hunk 1 above)
Comment (JohnMarksRH) [inline source/documentation/index.html.haml]: @duck-rh, got it

File: source/documentation/index.html.haml (hunk 2 above)
Comment (JohnMarksRH) [inline source/documentation/index.html.haml]: @duck-rh , good catch. Corrected

File: source/documentation/index.html.haml (hunk 1 above)
Comment (duck-rh) [inline source/documentation/index.html.haml]: You edited my comment by mistake :-/

Anyway, there is a character just before the small 'r' of "resources". It looks to me as a very small dot at mid-vertical position.

It seems to be this one:
U+05BC ( ◌ּ ) {HEBREW POINT DAGESH OR MAPIQ}

**Comment by JohnMarksRH** [comment]
@duck-rh @mykaul Typos corrected, one small change - 'How To > changed to "How to guides"

**Pull request merged**